src/processing/evm.rs

```diff
@@ -0,0 +1,198 @@
+use std::sync::Arc;
+use std::time::{SystemTime, UNIX_EPOCH};
+
+use alloy::hex::FromHex;
+use alloy::primitives::FixedBytes;
+use arc_swap::Guard;
+use bigdecimal::ToPrimitive;
+use num_bigint::BigInt;
+use starknet::core::types::{BlockId, BlockTag, Felt, FunctionCall};
+use starknet::macros::selector;
+use starknet::providers::jsonrpc::HttpTransport;
+use starknet::providers::{JsonRpcClient, Provider};
+
+use crate::config::{Config, EvmConfig};
+use crate::constants::EVM_TIME_SINCE_LAST_FEED_UPDATE;
+use crate::{config::get_config, error::MonitoringError};
+
+#[derive(Clone)]
+enum FeedType {
+    Unique(UniqueVariant),
+    Twap(TwapVariant),
+}
+
+impl FeedType {
+    async fn update_data(&self, chain: &EvmConfig, feed_id: FeedId) -> Result<(), MonitoringError> {
```

Comment:
should be implemented on `Feed` ;
```rust
let feed_id = felt!("0xTHE_FEED_ID");
let feed = Feed::from_id(feed_id);
let latest_data = feed.get_latest_data(evm_chain: &EvmConfig);
```

Reply:
(The current code is fine but we really want the same interface as the Cairo code, no parallel designs)